internal/net/grpc/client.go

```diff
@@ -681,6 +681,25 @@ func (g *gRPCClient) OrderedRangeConcurrent(
 	return nil
 }
 
+func RoundRobin[R any](
+	c Client,
+	ctx context.Context,
+	f func(ctx context.Context,
+		conn *grpc.ClientConn, copts ...grpc.CallOption) (R, error),
+) (data R, err error) {
+	res, err := c.RoundRobin(ctx, func(ctx context.Context, conn *grpc.ClientConn, copts ...grpc.CallOption) (any, error) {
+		return f(ctx, conn, copts...)
+	})
+	var zero R
```

Comment:
R is not always `nil`able in the type system and we cannot easily define `nil`able type. This initialization instantiates `nil` when `nil`able, default values when non`nil`able, so that we can return default or `nil` values when `err` is not `nil`.

Reply:
i think you can use data (named return value)